test/library/draft/Vector/Vector.chpl

```diff
@@ -0,0 +1,1264 @@
+/*
+ * Copyright 2020 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+  This module contains the implementation of the vector type.
+
+  Vector is a kind of list storing elements in a contiguous area.
+*/
+module Vector {
+
+  import ChapelLocks;
+  private use HaltWrappers;
+  private use Sort;
+  private use List only list;
+
+  pragma "no doc"
+  private const _initialCapacity = 8;
+  //
+  // We can change the lock type later. Use a spinlock for now, even if it
+  // is suboptimal in cases where long critical sections have high
+  // contention (IE, lots of tasks trying to insert into the middle of this
+  // vector, or any operation that is O(n)).
+  //
+  pragma "no doc"
+  type _lockType = ChapelLocks.chpl_LocalSpinlock;
+
+  //
+  // Use a wrapper class to let vector methods have a const ref receiver even
+  // when `parSafe` is `true` and the vector lock is used.
+  //
+  pragma "no doc"
+  class _LockWrapper {
+    var lock$ = new _lockType();
+
+    inline proc lock() {
+      lock$.lock();
+    }
+
+    inline proc unlock() {
+      lock$.unlock();
+    }
+  }
+
+  private use IO;
+
+  proc _checkType(type t) {
+    if (!isDefaultInitializable(t)) {
+      compilerError("Vector does not support an eltType that can't be default initialized, ", 
+                    "here: ", t:string);
+    }
+  }
+  record vector {
+    /* The type of the elements contained in this vector. */
+    type eltType;
+    /* If `true`, this vector will perform parallel safe operations. */
+    param parSafe = false;
+
+
+    pragma "no doc"
+    var _size = 0;
+
+    pragma "no doc"
+    var _capacity = 0;
```

Comment:
It'd be nice to add comments for these variables. Especially size vs capacity

Reply:
https://github.com/chapel-lang/chapel/pull/16048/commits/6d2dfefa3fb1978f9716bed43c2b6b7f7d4fcc77

Reply:
Do you have any doubts about the description I added? Or I will mark the conversation as resolved.

Reply:
Nope, looks good.